src/functions/exportFunctions.js

```diff
@@ -0,0 +1,115 @@
+import { port } from 'App';
+import reactJsLogo from '../assets/images/react-logo-svgrepo-com.svg';
+import jsLogo from '../assets/images/js.svg';
+import nodeJsLogo from '../assets/images/nodejs.svg';
+import dataStructureLogo from '../assets/images/hierarchical-structure-svgrepo-com.svg';
+
+export const backendApiCall = async (method, url, body) => {
+  const options = {
+    method: method,
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    credentials: 'include',
+  };
+
+  if (method !== 'GET' && body) {
+    options.body = JSON.stringify(body);
+  }
+
+  try {
+    const response = await fetch(`${port}${url}`, options);
+
+    if (!response.ok) {
+      const errorMessage = `Error: ${response.status} - ${response.statusText}`;
+      throw new Error(errorMessage);
+    }
+
+    const contentType = response.headers.get('content-type');
+    if (contentType && contentType.includes('application/json')) {
+      return await response.json();
+    } else {
+      return await response.text();
+    }
+  } catch (error) {
+    throw error;
+  }
+};
+
+export const authenticateUser = async (backendApiCall, setAuth, setLoading) => {
+  try {
+    const backendUserData = await backendApiCall('GET', '/api/v1/login');
+    setAuth({
+      userId: backendUserData.user.userId,
+      firstName: backendUserData.user.firstname,
+      lastName: backendUserData.user.lastname,
+      email: backendUserData.user.url,
+      role: backendUserData.user.role,
+      loggedIn: true,
+      accessToken: backendUserData.user.accessToken,
+      avatarURL: backendUserData.user.avatarURL,
+      isActive: backendUserData.isActive,
+    });
+  } catch (error) {
+    throw new Error(error);
+  } finally {
+    setLoading(false);
+  }
+};
+
+export const handleLogout = async (
+  backendApiCall,
+  setAuth,
+  setShowLogoutModal
+) => {
+  try {
+    await backendApiCall('GET', '/api/v1/logout');
+
+    setAuth({
+      loggedIn: false,
+    });
+    setShowLogoutModal(true);
+  } catch (error) {
+    // Handle any errors here, such as showing an error message
+    console.error('Logout failed:', error);
+  }
+};
+
+export const fetchAndTransformQuizzes = async (
+  backendApiCall,
+  setQuizzes,
```

Comment:
To make API handling functions more extensible I'd suggest naming this callback something more general like `onSuccess`. This way if you ever have to do something more general than setting state after the API call `fetchAndTransformedOptions` it'll be apparent which callback to use.

Reply:
Cool, good suggestion.  Also, on looking at this function a little more closely, I thought it would be a good idea to add some error handling as well.  Thanks!